File: app/src/main/java/com/android/learn/mcontract/KnowledgeChildContract.java
package com.android.learn.mcontract;


import com.android.base.mmodel.FeedArticleListData;
import com.android.base.mmodel.FeedArticleListData.FeedArticleData;


public class KnowledgeChildContract {

    public interface Presenter {
        void getKnowledgeArticleList(int num,int cid);

        void onRefreshMore(int cid);

        void onLoadMore(int cid);

        void addCollectArticle(int position, FeedArticleData feedArticleData);

        void cancelCollectArticle(int position, FeedArticleData feedArticleData);
    }

    public interface View {
        void showArticleList(FeedArticleListData itemBeans, boolean isRefresh);
        void showCollectArticleData(int position, FeedArticleData feedArticleData);

        void showCancelCollectArticleData(int id, FeedArticleData feedArticleData);
    }

}

File: app/src/main/java/com/android/learn/mcontract/KnowledgeChildContract.kt
package com.android.learn.mcontract


import com.android.base.mmodel.FeedArticleListData
import com.android.base.mmodel.FeedArticleListData.FeedArticleData
import com.android.base.mview.BaseView


class KnowledgeChildContract {

    interface Presenter {
        fun getKnowledgeArticleList(num: Int, cid: Int)

        fun onRefreshMore(cid: Int)

        fun onLoadMore(cid: Int)

        fun addCollectArticle(position: Int, feedArticleData: FeedArticleData)

        fun cancelCollectArticle(position: Int, feedArticleData: FeedArticleData)
    }

    interface View : BaseView {
        fun showArticleList(itemBeans: FeedArticleListData, isRefresh: Boolean)
        fun showCollectArticleData(position: Int, feedArticleData: FeedArticleData)

        fun showCancelCollectArticleData(id: Int, feedArticleData: FeedArticleData)
    }

}
